mule-deployer/src/main/java/org/mule/tools/client/core/logging/ClientLoggingFilter.java

```diff
@@ -102,28 +104,23 @@ public void aroundWriteTo(WriterInterceptorContext writerInterceptorContext) thr
     }
   }
 
-  private class RequestLoggingStream extends FilterOutputStream {
-
-    private StringBuilder request;
-    private ByteArrayOutputStream requestBody = new ByteArrayOutputStream();
-
-    RequestLoggingStream(StringBuilder request, OutputStream inner) {
-      super(inner);
-      this.request = request;
+  private void appendHeaders(StringBuilder builder, MultivaluedMap<String, String> headers) {
+    for (Map.Entry<String, List<String>> headerEntry : headers.entrySet()) {
+      builder.append(headerEntry.getKey()).append(": ").append(join(headerEntry.getValue(), ", ")).append(format("%n"));
     }
+    builder.append(format("%n"));
+  }
 
-    @Override
-    public void write(final int i) throws IOException {
-      requestBody.write(i);
-      out.write(i);
+  private boolean shouldLogEntity(Type entityType) {
+    if (!entityType.getTypeName().contains("FormDataMultiPart")) {
+      return true;
     }
 
+    return entityType.getTypeName().contains("FormDataMultiPart") && shouldLogMultiPart();
+  }
 
-    public String getRequestLog() {
-      request.append(new String(requestBody.toByteArray()));
-      request.append('\n');
-      return request.toString();
-    }
-
+  private Boolean shouldLogMultiPart() {
+    return parseBoolean(System.getProperty(CLIENT_LOGGING_LOG_MULTIPART, "false"));
```

Comment:
Just if you are interested to there is a Boolean.getBoolean(yourSystemProperty)

Reply:
yep good call done